### PR TITLE
plugin: container images registry

### PR DESCRIPTION
**Pull request by danielerez**
Added a plugin for connecting to an image registry server.
The plugin adds a tab to 'Connect to server' dialog:
* Server: e.g. docker.io
* Image: e.g. cirros
* Type: docker, oci or local dir
* Credentials (if required)

Upon connection 'skopeo inspect' is invoked to get the image's layers (each layer is a .tar.gz file).
Then, 'skopeo copy' is invoked to pull the image into a temp directory.

### CODE REVIEW

File: mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/AbstractRegistryImage.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	}
+
+	protected RegistryConnHandler getConnHandler() throws IOException {
+		RegistryConnHandler connection = (RegistryConnHandler) ConnectionPool
+				.getConnectionHandler(this, fileURL, true);
+		try {
+			connection.checkConnection();
+		} catch (IOException e) {
+			throw e;
+		}
```
Comment (ahadas) [inline mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/AbstractRegistryImage.java]: was it used to for debugging?

File: mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/AbstractRegistryImage.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+	@Override
+	public AbstractFile getParent() {
+		if (parent == null) {
+			FileURL parentFileURL = this.fileURL.getParent();
+            if(parentFileURL!=null) {
+                parent = FileFactory.getFile(parentFileURL);
+                // Note: parent may be null if it can't be resolved
+            }
+		}
+		return parent;
```
Comment (ahadas) [inline mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/AbstractRegistryImage.java]: formatting

File: mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/SkopeoCommandExecutor.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        return tempFolder;
+    }
+
+    public static void execute(StringBuilder outputBuffer, String command, Credentials creds, String... args) throws IOException {
+        LOGGER.debug("Executing skopeo command: " + command);
+        List<String> tokens = new ArrayList<>(Arrays.asList(
+                "skopeo",
+                "--insecure-policy",
+                "--override-os",
+                "linux",
```
Comment (ahadas) [inline mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/SkopeoCommandExecutor.java]: should this be set to the running operating system?

File: build.gradle [hunk 4]
```diff
@@ -16,6 +16,8 @@ dependencies {
 
     osgiRuntime project('mucommander-sftp')
     osgiRuntime project('mucommander-rar')
+    osgiRuntime project('mucommander-ovirt')
```
Comment (ahadas) [inline build.gradle]: the ovirt plugin doesn't seem to be ready yet - can you please rebase on top of the master branch?

File: mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/RegistryImage.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import com.mucommander.commons.file.FileFactory;
+import com.mucommander.commons.file.archive.tar.TarArchiveFile;
+import com.mucommander.commons.file.util.PathUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mucommander.commons.file.AbstractFile;
+import com.mucommander.commons.file.FileURL;
+
+public class RegistryImage extends AbstractRegistryImage {
```
Comment (ahadas) [inline mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/RegistryImage.java]: correct me if I'm wrong - but that kind of mixes the protocol and the archive together, right?
wouldn't it make more sense to separate them out? meaning, to have this protocol-plugin listing, downloading and uploading images from the repositories and have an archive plugin for manipulating the images?

File: mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/AbstractRegistryImage.java (hunk 1 above)
Comment (danielerez) [reply mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/AbstractRegistryImage.java]: yeah, removed.

File: mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/AbstractRegistryImage.java (hunk 2 above)
Comment (danielerez) [reply mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/AbstractRegistryImage.java]: done

File: mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/SkopeoCommandExecutor.java (hunk 3 above)
Comment (danielerez) [reply mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/SkopeoCommandExecutor.java]: It's for supporting downloading linux images under other OSes. In the future, we probably should add a select-box for choosing an OS.

File: build.gradle (hunk 4 above)
Comment (danielerez) [reply build.gradle]: sure. will do.

File: mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/RegistryImage.java (hunk 5 above)
Comment (danielerez) [reply mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/RegistryImage.java]: no need, as we don't actually listing images, but rather downloading a specific image and display its layers.

**Comment by ahadas** [comment]
@danielerez Thanks for this contribution!

**Pull request merged**